modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/util/IndexCreatorIfNotPresent.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.util;
+
+import com.expedia.adaptivealerting.modelservice.dao.es.ElasticSearchClient;
+import com.expedia.adaptivealerting.modelservice.dao.es.ElasticSearchConfig;
+import com.google.gson.JsonObject;
+import lombok.Generated;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.admin.indices.create.CreateIndexResponse;
+import org.elasticsearch.action.admin.indices.get.GetIndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import static com.expedia.adaptivealerting.modelservice.dao.es.DetectorMappingEntity.CREATE_TIME_KEYWORD;
+import static com.expedia.adaptivealerting.modelservice.dao.es.DetectorMappingEntity.DETECTOR_KEYWORD;
+import static com.expedia.adaptivealerting.modelservice.dao.es.DetectorMappingEntity.LAST_MOD_TIME_KEYWORD;
+import static com.expedia.adaptivealerting.modelservice.dao.es.DetectorMappingEntity.QUERY_KEYWORD;
+import static com.expedia.adaptivealerting.modelservice.dao.es.DetectorMappingEntity.USER_KEYWORD;
+
+/**
+ * Util class to create index with mappings if not found.
+ */
+@Component
+@Slf4j
+@Generated //(exclude from code coverage)
```

Comment:
This doesn't feel like a legitimate exclusion target to me. Can we discuss further please?

Reply:
Updated.

Reply:
Did you add a unit test? I don't see it.

Reply:
@dinilatgit ^

Reply:
Sorry, missed information in the comment - mock for final method not supported.